lib/agent/redirect.js

```diff
@@ -16,33 +13,9 @@ class RedirectHandler {
     this.destroyed = false
   }
 
-  destroy (err) {
-    this.destroyed = true
-
-    if (this.abort) {
-      this.abort()
-    }
-
-    this.handler.onError(err)
-  }
-
-  get (origin) {
-    return this[kAgent].get(origin)
-  }
-
   onConnect (abort) {
-    if (this.destroyed) {
-      abort()
-    } else {
-      this.abort = abort
-      this.handler.onConnect(abort)
-    }
-  }
-
-  onUpgrade (statusCode, headers, socket) {
-    assert(!this.location, 'redirect should not upgrade')
-
-    this.handler.onUpgrade(statusCode, headers, socket)
+    this.abort = abort
+    this.handler.onConnect(abort)
   }
```

Comment:
All if these can happen. Please put them back.

Reply:
Done. I was not able to find in the code how to reach those hooks so I leave it up to you.